ion-schema/src/system.rs

```diff
@@ -754,37 +753,79 @@ impl Resolver {
         let mut isl_inline_imports: Vec<IslImportType> = vec![];
         let mut open_content = vec![];
         let mut isl_user_reserved_fields = UserReservedFields::default();
+        let mut isl_version = IslVersion::V1_0;
+        let isl_version_marker: Regex = Regex::new(r"^\$ion_schema_\d.*$").unwrap();
+        let reserved_keyword_version_marker =
+            Regex::new(r"^(\$ion_schema(_.*)?|[a-z][a-z0-9]*(_[a-z0-9]+)*)$").unwrap();
```

Comment:
Can we move these to a static [`OnceLock`](https://doc.rust-lang.org/std/sync/struct.OnceLock.html)? E.g.
```rust
static ISL_VERSION_MARKER_REGEX: OnceLock<Regex> = OnceLock::new();

/// Checks if a value is an ISL version marker.
fn is_isl_version_marker(text: &str) -> bool {
    ISL_VERSION_MARKER_REGEX
        .get_or_init(|| Regex::new(r"^\$ion_schema_\d.*$").unwrap())
        .is_match(text)
}
```

Alternately, we can create a new issue to update the Rust version and deal with all of the regexes.

Reply:
I have used `OnceLock<Regex>` here.